tests/core/util/test_keychain.py

```diff
@@ -165,6 +166,28 @@ def test_bip39_test_vectors(self):
             assert bytes_to_mnemonic(entropy_bytes) == mnemonic
             assert mnemonic_to_seed(mnemonic) == seed
 
+    def test_bip39_test_vectors_short(self):
+        """
+        Tests that the first 4 letters of each mnemonic phrase matches as if it were the full phrase
+        """
+        with open("tests/util/bip39_test_vectors_short.json") as f:
+            all_vectors = json.loads(f.read())
+
+        with open("tests/util/bip39_test_vectors.json") as f:
+            full_vectors = json.loads(f.read())
+
+        for idx, vector_list in enumerate(all_vectors["english"]):
```

Comment:
I would probably unpack here first and avoid 0, 1, 2...  but yeah, it is still index based access so...  not a biggy.

```suggestion
        for idx, [this, that, the other] in enumerate(all_vectors["english"]):
```

Parallel lists are also icky, but when you have them you can zip.

```suggestion
        for idx, [av, fv] in zip(all_vectors["english"], full_vectors["english"]):`
```

Where that can be combined with unpacking like `for index, [[e, m, s], [_, f, _]] in...`.  Or...  using good names of course.

Reply:
Updated in last commit